### ISSUE TITLE
Nito reference is out of date

### ISSUE THREAD

**Issue by robertsimmons** [box/box-windows-sdk-v2#112]
Nito is up to version 3.x - https://www.nuget.org/packages/Nito.AsyncEx/

Is it possible to get a new version of this SDK that uses the latest version? As it currently stands, if you update Nito through NuGet, the Box SDK will no longer work (System.IO.FileLoadException: Could not load file or assembly 'Nito.AsyncEx, Version=2.1.3.0, Culture=neutral, PublicKeyToken=09d695329ca273b7' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)). The Box dependency says Nito >= 2.1.3 should work, but obviously over 3 is failing. Strongly named reference I'm assuming?

**Comment by judahsali** [comment]
Same issue here. Can't use the previous version of Nito.AsyncEx because of a shared dependency on Microsoft.Bcl.Async. Help please.

**Comment by cburnette** [comment]
I'll look into updating to latest. 

Chad

**Comment by cburnette** [comment]
Nito 3.x is not signed (i.e. no Strong Name), which means it doesn't work with the rest of the project.  If they get around to signing it we can upgrade.

**Comment by judahsali** [comment]
I don't think they plan on signing it. See here: https://github.com/StephenCleary/AsyncEx#strong-naming

**Comment by imagine4me** [comment]
Stop using it then??!? It's a trivial library...

**Comment by gmichaud** [comment]
The Box Windows SDK makes very little use of NitoEx. We actually forked it to reduce the number of external dependencies (all we need is support of .NET 4.5, no PCL needed in our project and I didn't like having to ship all the BCL stuff with it). It was not trivial to extract the relevant bits of code from NitoEx and we simply replaced the AsyncEx code with http://blogs.msdn.com/b/pfxteam/archive/2012/02/12/10266988.aspx. Our fork is at https://github.com/Acumatica/acumatica-box-windows-sdk-v2.
